tests/cluster/tests/28-cluster-shards.tcl

```diff
@@ -285,3 +285,24 @@ test "CLUSTER MYSHARDID reports same shard id after cluster restart" {
         assert_equal [dict get $node_ids $i] [R $i cluster myshardid]
     }
 }
+
+test "Deterministic order of CLUSTER SHARDS response" {
```

Comment:
This test isn't part of the CI since it uses the legacy clustering test framework. We should either move cluster-shards to the new framework or just move this part of the file over and we can remove the rest of the file in a separate PR.